java-spiffe-provider/src/main/java/io/spiffe/provider/SpiffeIdVerifier.java

```diff
@@ -0,0 +1,17 @@
+package io.spiffe.provider;
+
+import io.spiffe.spiffeid.SpiffeId;
+
+import java.security.cert.X509Certificate;
+
+public interface SpiffeIdVerifier {
+    /**
+     * Verify that an X509-SVID is acceptable. This method receives the SPIFFE ID of the SVID and the certificate
+     * chain.
+     *
+     * @param spiffeId the SPIFFE ID of the SVID
+     * @param verifiedChain the certificate chain with the X509-SVID certificate back to an X.509 root for the trust domain.
+     * @return true if the SPIFFE ID is acceptable
+     */
+    public boolean verify(SpiffeId spiffeId, X509Certificate[] verifiedChain);
```

Comment:
I feel that this should throw an exception rather than return a boolean. Any standard certificate validation procedures in Java using `X509TrustManager` implementations (i.e. `checkClientTrusted()`, `checkServerTrusted()` methods) will throw an exception.

It feels unnatural to swallow such validation exceptions inside the implementation of this interface and compress the result to a single boolean that doesn't indicate the reason why the verification failed.

Reply:
I'm fine doing a throw instead. I was just following the pattern of [HostnameVerifier.verify](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/javax/net/ssl/HostnameVerifier.html#verify(java.lang.String,javax.net.ssl.SSLSession)) for SpiffeIdVerifier.verify.